specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,248 @@
+
+Of the 44 UWP activation kinds, 5 of the most commonly-used kinds are also supported for Desktop Bridge and Windows [reg-free WinRT](https://blogs.windows.com/windowsdeveloper/2019/04/30/enhancing-non-packaged-desktop-apps-using-windows-runtime-components/) apps. The AppLifecycle implementation is based on this set. Some activation kinds require prior registration, others do not. The target list for v1 is as follows:
+
+|     Activation   kind    |     Description                                                                                                                                             |
+|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|     Launch               |     Launch   from a Start tile/shortcut, from a command window, or programmatically via   ShellExecute/CreateProcess.                                       |
+|     File                 |     The   app is activated when the user double-clicks a registered file, or when a   caller calls ShellExecute or LaunchFileAsync on a registered file.    |
+|     Protocol             |     Activated   via ShellExecute or LaunchUriAsync, or by specifying a protocol string on a   command-line.                                                 |
+|     StartupTask          |     The   app starts on user log-in - either registered in the registry, or via   shortcuts in a well-known startup folder.                                 |
+|     ToastNotification    |     The   app sends a toast, the user responds, and this activates the app.                                                                                 |
+    
+
+### Activation registration
+From a Win32 app perspective, the 2 activation kinds that do not require any registration (Shortcut/Tile Launch and CommandLine) are effectively the same in terms of activation payload. Indeed, it is not possible to distinguish a launch activation of a Win32 app from a commandline activation. Therefore, the AppLifecycle implementation will merge Launch and CommandLine into one (Launch) activation kind. 
+
+UWP apps can only register for activation kinds via manifest declarations, and therefore this happens during app install. Conversely, traditional app types can write registry entries at any time.
+
+A key requirement is to enable unpackaged Win32 apps to use the set of rich activations using familiar designs, without any additional friction. Specifically, AppLifecycle will provide APIs which the app can call to register its interest in rich activations. The app is not required to have an MSIX manifest, nor an MSIX package. Using the API approach rather than the manifest approach has the added benefit that it also addresses a related customer request for dynamic activation registrations even for UWP apps.
```

Comment:
It's a requirement on us, but what's this give to developers?

Reply:
Updated.